### ISSUE TITLE
Changing hyper-parameters in PPO2 

### ISSUE THREAD

**Issue by meric-sakarya** [araffin/rl-baselines-zoo#94]
I am trying to change the hyperparameters of PPO2 to achieve better results in Pendulum-v0. I have tried using the parameters from RL Baselines Zoo [here](https://github.com/araffin/rl-baselines-zoo/blob/master/hyperparams/ppo2.yml#L14) but I receive this error.
```
Traceback (most recent call last):
  File "C:/Users/meric/OneDrive/Masaüstü/TUM/Thesis/Pycharm/pioneer/pendulum_default_PPO2.py", line 35, in <module>
    cliprange=0.2, env=pendulum_env, verbose=1, tensorboard_log="./ppo2_pendulum_default_tensorboard/")
TypeError: __init__() got an unexpected keyword argument 'n_envs'
```

Here is the code: 
```
import os
import time

import gym
from gym import Wrapper, spaces
import numpy as np
from gym.envs.classic_control import PendulumEnv

from stable_baselines.common.env_checker import check_env
from stable_baselines.sac.policies import MlpPolicy
from stable_baselines import PPO2
from stable_baselines.common.vec_env import DummyVecEnv
from stable_baselines.bench import Monitor

import tensorflow as tf

# tensorboard --logdir=PPO2_DEFAULT_PENDULUM:C:\Users\meric\OneDrive\Masaüstü\TUM\Thesis\Pycharm\pioneer\ppo2_pendulum_default_tensorboard --host localhost

log_dir = "/tmp/gym/{}".format(int(time.time()))
os.makedirs(log_dir, exist_ok=True)

config = tf.ConfigProto()

config.gpu_options.allow_growth = True
sess = tf.Session(config=config)

TEST_COUNT = 100

pendulum_env = gym.make('Pendulum-v0')
pendulum_env = Monitor(pendulum_env, log_dir, allow_early_resets=True)
check_env(pendulum_env, warn=True)

model = PPO2(n_envs=8, n_timesteps=2e6, policy='MlpPolicy', n_steps=2048, nminibatches=32, lam=0.95, gamma=0.99,
             noptepochs=10, ent_coef=0.0, learning_rate=3e-4,
             cliprange=0.2, env=pendulum_env, verbose=1, tensorboard_log="./ppo2_pendulum_default_tensorboard/")
model.learn(total_timesteps=100_000, log_interval=10)
model.save("ppo2_pendulum_default")

```

System Info
Describe the characteristic of your environment:

Describe how stable baselines was installed (pip, docker, source, ...): source
GPU models and configuration: NVIDIA GTX 1050 with CUDA 10.0 and cuDNN 7.6.5
Python version: 3.7
Tensorflow version: 1.15

**Comment by araffin** [comment]
Hello,

Why are you not using the zoo?
Please read the documentation to understand the meaning of each variable (`n_envs` correspond to the number of environments/workers to collect data). You can also read the code from the zoo `train.py` and `utils/utils.py` to see how `n_envs` is used.

**Comment by meric-sakarya** [comment]
I don't know how to install zoo on Windows and couldn't find documentation on it.

**Comment by araffin** [comment]
>I don't know how to install zoo on Windows and couldn't find documentation on it.

You just need to clone the repo and install python dependencies, it is not specific to windows.
You could also use it online using google colab (cf README).

**Comment by meric-sakarya** [comment]
Can't I just use the hyperparameters provided by zoo without actually using zoo? I am not interested in already trained agents.

**Comment by araffin** [comment]
>Can't I just use the hyperparameters provided by zoo without actually using zoo? I am not interested in already trained agents and also I am going to need some wrappers in another environment.

In SB3 version, you can download only the training scripts without the trained agents.
Here, you can download the repo and then delete the `trained_agents` folder.
Otherwise, you can just copy the files.

**Comment by meric-sakarya** [comment]
I managed to get an agent working thanks for the help. But isn't this [graph](https://ibb.co/9Y1Wv41) pretty weird? What might be causing this? Can I wrap this with the monitor wrapper to get a better looking graph?

**Comment by araffin** [comment]
There are already issues about that: https://github.com/hill-a/stable-baselines/issues/143
In short: you should ignore it and use Monitor wrapper or proper evaluation (using `EvalCallback` included in the zoo).

Closing this issue as the original questions was answered.

**Comment by meric-sakarya** [comment]
I added the following line to the ppo2.yml file: env_wrapper: stable_baselines.bench.monitor. It did not work and I received his error when trying to restart the training: 
```
(pioneer) C:\Users\meric\OneDrive\Masaüstü\TUM\Thesis\Zoo\rl-baselines-zoo>python train.py --algo ppo2 --env Pendulum-v0 --tensorboard-log C:\Users\meric\OneDrive\Masaüstü\TUM\Thesis\Zoo                                                      2020-07-25 16:12:40.012000: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll                                                                                                WARNING:tensorflow:                                                                                                     The TensorFlow contrib module will not be included in TensorFlow 2.0.                                                   For more information, please see:                                                                                         * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md                                   * https://github.com/tensorflow/addons                                                                                  * https://github.com/tensorflow/io (for I/O related ops)                                                              If you depend on functionality not listed there, please file an issue.                                                                                                                                                                          WARNING:tensorflow:From C:\Users\meric\Anaconda3\envs\pioneer\lib\site-packages\stable_baselines\common\misc_util.py:26: The name tf.set_random_seed is deprecated. Please use tf.compat.v1.set_random_seed instead.                                                                                                                                                    ========== Pendulum-v0 ==========                                                                                       Seed: 0                                                                                                                 OrderedDict([('cliprange', 0.2),                                                                                                     ('ent_coef', 0.0),                                                                                                      ('env_wrapper', 'stable_baselines.bench.monitor'),                                                                      ('gamma', 0.99),                                                                                                        ('lam', 0.95),                                                                                                          ('learning_rate', 0.0003),                                                                                              ('n_envs', 8),                                                                                                          ('n_steps', 2048),                                                                                                      ('n_timesteps', 2000000.0),                                                                                             ('nminibatches', 32),                                                                                                   ('noptepochs', 10),                                                                                                     ('policy', 'MlpPolicy')])                                                                                  Using 8 environments                                                                                                    Traceback (most recent call last):                                                                                        File "train.py", line 284, in <module>                                                                                    env = create_env(n_envs)                                                                                              File "train.py", line 262, in create_env                                                                                  wrapper_class=env_wrapper, env_kwargs=env_kwargs) for i in range(n_envs)])                                            File "C:\Users\meric\Anaconda3\envs\pioneer\lib\site-packages\stable_baselines\common\vec_env\dummy_vec_env.py", line 23, in __init__                                                                                                             self.envs = [fn() for fn in env_fns]                                                                                  File "C:\Users\meric\Anaconda3\envs\pioneer\lib\site-packages\stable_baselines\common\vec_env\dummy_vec_env.py", line 23, in <listcomp>                                                                                                           self.envs = [fn() for fn in env_fns]                                                                                  File "C:\Users\meric\OneDrive\Masaüstü\TUM\Thesis\Zoo\rl-baselines-zoo\utils\utils.py", line 175, in _init                env = wrapper_class(env)                                                                                              File "C:\Users\meric\OneDrive\Masaüstü\TUM\Thesis\Zoo\rl-baselines-zoo\utils\utils.py", line 141, in wrap_env             env = wrapper_class(env, **kwargs)                                                                                  TypeError: 'module' object is not callable                                                                                                                                                  ```
